pandas/_libs/tslibs/strptime.pyx

```diff
@@ -632,3 +655,48 @@ cdef _calc_julian_from_U_or_W(int year, int week_of_year,
     else:
         days_to_week = week_0_length + (7 * (week_of_year - 1))
         return 1 + days_to_week + day_of_week
+
+cdef _parse_timezone_directive(object z):
+    """
+    Parse the '%z' directive and return an offset from UTC
+
+    Parameters
+    ----------
+    z : string-like of the UTC offset
+
+    Returns
+    -------
+    tuple of (offset, offset_fraction)
+    """
+
+    cdef:
+        int gmtoff, gmtoff_fraction, hours, minutes, seconds, pad_number
+        object gmtoff_remainder, gmtoff_remainder_padding
+
+    if z == 'Z':
+        gmtoff = 0
+        gmtoff_fraction = 0
+    else:
+        if z[3] == ':':
```

Comment:
you might need to wrap this entire block in a try/except if the string is not long enough (or check lengths for each sub-section) and the raise the appropriate error